packages/babel-parser/src/plugins/flow.js

```diff
@@ -901,9 +936,41 @@ export default (superClass: Class<Parser>): Class<Parser> =>
           );
         }
         this.expect(tt.ellipsis);
-        node.argument = this.flowParseType();
+        switch (this.state.type) {
+          case tt.comma:
+          case tt.semi:
+          case tt.braceR:
+          case tt.braceBarR: {
```

Comment:
I think that you can also remove this outer `switch`, since the `if`s already handle only these four cases.